pkg/recordset/recordset.go

```diff
@@ -142,8 +150,24 @@ func getStackNames(cl client.StackLister, re *regexp.Regexp) ([]string, error) {
 	var result []string
 
 	for _, item := range output.StackSummaries {
+		// filter stack by name.
 		if re.Match([]byte(*item.StackName)) {
-			result = append(result, *item.StackName)
+			// filter stack by installation tag.
+			describeInput := &cloudformation.DescribeStacksInput{
+				StackName: aws.String(*item.StackId),
+			}
+			stacks, err := cl.DescribeStacks(describeInput)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			for _, stack := range stacks.Stacks {
```

Comment:
It is in general a good idea to separate dependencies and algorithms. This here is now all together and not easily testable.

Reply:
Moved out validation part to own functions. PTAL